### PR TITLE
Add missing target settings `MERGED_BINARY_TYPE` and `MERGEABLE_LIBRARY`

### PR DESCRIPTION
**Pull request by woin2ee**
Resolves <https://github.com/tuist/tuist/issues/6348>

### Short description 📝

It makes sure that `MERGED_BINARY_TYPE` and `MERGEABLE_LIBRARY` settings are configured correctly.

### How to test the changes locally 🧐

Clone https://github.com/dmytro-anokhin/tuist-default-settings-essental and run `tuist generate`.
Confirm `MERGED_BINARY_TYPE` on **App** target and `MERGEABLE_LIBRARY` on **Framework** target equal to settings of manifest.

### Contributor checklist ✅

- [x] The code has been linted using run `mise run lint-fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [ ] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

**Review by fortmarek — APPROVED** [review]
looks good! I will rerun the failed pipeline as it looks unrelated to your changes and the dependencies acceptance tests have been somewhat flaky in the near past.

**Comment by fortmarek** [comment]
The `BuildAcceptanceTestMultiplatformAppWithMacrosAndEmbeddedWatchOSApp` failed on a subsequent re-run, would you mind checking it locally if you can reproduce it?

**Comment by woin2ee** [comment]
> The BuildAcceptanceTestMultiplatformAppWithMacrosAndEmbeddedWatchOSApp failed on a subsequent re-run, would you mind checking it locally if you can reproduce it?

I'm sorry, I missed the failed test case because macro-related test cases always failed locally.

Anyway, the reason of failure is that `MERGED_BINARY_TYPE` and `MERGEABLE_LIBRARY` are not supported for architecture `i386`. I'm not sure why exactly, but `i386` is being used in the build time if these options are set on external dependency targets.

I've experimented by creating a new project then Xcode doesn't add anything to the `project.pbxproj` file if I haven't touch these options.(It showed "NO") So, I followed Xcode's style for avoid problems such as it.

File: Sources/TuistGenerator/Generator/ConfigGenerator.swift [hunk 1]
```diff
@@ -285,6 +285,23 @@ final class ConfigGenerator: ConfigGenerating {
 
         settings["PRODUCT_NAME"] = .string(target.productName)
 
+        // Actually, when `MERGEABLE_LIBRARY` is NO and project is generated, Xcode doesn't include the line of setting in
+        // `project.pbxproj` file.
```
Comment (fortmarek) [inline Sources/TuistGenerator/Generator/ConfigGenerator.swift:289]:
```suggestion
```

File: Sources/TuistGenerator/Generator/ConfigGenerator.swift [hunk 2]
```diff
@@ -285,6 +285,23 @@ final class ConfigGenerator: ConfigGenerating {
...
+        if target.mergeable {
+            settings["MERGEABLE_LIBRARY"] = .string("YES")
+        }
+
+        switch target.mergedBinaryType {
+        case .disabled:
+            // Actually, when `MERGED_BINARY_TYPE` is disabled and project is generated, Xcode doesn't include the line of setting
+            // in `project.pbxproj` file.
```
Comment (fortmarek) [inline Sources/TuistGenerator/Generator/ConfigGenerator.swift:297]:
```suggestion
            // When `MERGED_BINARY_TYPE` is disabled, `MERGED_BINARY_TYPE` value should be left empty
```

**Review by fortmarek — APPROVED** [review]

**Comment by fortmarek** [comment]
@woin2ee would you mind pulling the latest changes from `main`, so this [PR](https://github.com/tuist/tuist/pull/6457) is included here? It should help to resolve the failure of the dependencies workflow.

**Comment by fortmarek** [comment]
great work @woin2ee 👏

**Pull request merged**